src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -54,4 +66,44 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     final protected String parserName() {
         return MatchAllQueryParser.NAME;
     }
-}
\ No newline at end of file
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        if (this.boost == 1.0f) {
+            return Queries.newMatchAllQuery();
+        }
+
+        MatchAllDocsQuery query = new MatchAllDocsQuery();
+        query.setBoost(boost);
+        return query;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+
+        MatchAllQueryBuilder other = (MatchAllQueryBuilder) obj;
+        return Objects.equals(boost, other.boost);
```

Comment:
Thinking about it, using "==" is probably not the best way for float primitives. The internal Float.equals() uses `floatToIntBits()` just like the hashCode(), maybe we should better use that for consitency?